src/screens/LearningScreen/components/FlashCardContainer.js

```diff
@@ -0,0 +1,121 @@
+import React, { useState, useEffect } from 'react';
+import { View, Button, Text } from 'react-native';
+
+import { getUserData } from '../../../utils/storage';
+import {
+  getRandomWord,
+  handleCorrectOrIncorrect,
+  getNewWordLevel,
+  LIMITS,
+  LEVELS,
+} from '../../../utils/spacing';
+import FlashCard from './FlashCard';
+
+const handleRevealButtonPress = (setRevealed) => {
+  setRevealed((flipped) => !flipped);
+};
+
+const handleExpressionsButtonPress = (setExpanded) => {
+  setExpanded((isExpanded) => !isExpanded);
+};
+
+const FlashCardContainer = () => {
+  const [list, setList] = useState({});
+  const [wordInfo, setWordInfo] = useState({});
+  const [isExpanded, setExpanded] = useState(false);
+  const [isRevealed, setRevealed] = useState(false);
+  const [currentLevel, setCurrentLevel] = useState('one');
+  const [numQuizzed, setNumQuizzed] = useState(0);
+
+  const handleCorrectOrIncorrectButtonPress = (isCorrect) => {
+    const temp = handleCorrectOrIncorrect(
+      isCorrect,
+      wordInfo,
+      list,
+      currentLevel,
+    );
+
+    // EFFECTS
+    setList(temp);
+    setNumQuizzed((quizzed) => quizzed + 1);
+    setRevealed(false);
+    setExpanded(false);
+  };
+
+  useEffect(() => {
+    let mounted = true;
+
+    getUserData().then((userData) => {
+      if (mounted) {
+        setList(userData.wordList);
+      }
+    });
+
+    return () => {
+      mounted = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!list || !currentLevel || !list[currentLevel]) {
```

Comment:
Something to watch out for: hypothetically if `currentLevel` _could_ be an integer with a value of 0 and that's considered valid, then a conditional such as this one could yield incorrect results (if the `!currentLevel` was intended to check whether it's defined or not defined). Because you're using the string `'one', 'two'`, etc. then this isn't a concern here. But if you were to take Doug's suggestion and change the `currentLevel` to be an integer you need to be aware of how that affects conditionals such as this one.

Reply:
Right, definitely something for me to be aware of. When I do make that change, I'll make it a 1-based index to avoid that issue and make it more readable for the user